app/models/order_cycle.rb

```diff
@@ -22,17 +22,23 @@ class OrderCycle < ApplicationRecord
 
   has_many :suppliers, -> { distinct }, source: :sender, through: :cached_incoming_exchanges
   has_many :distributors, -> { distinct }, source: :receiver, through: :cached_outgoing_exchanges
-
   has_many :order_cycle_schedules
   has_many :schedules, through: :order_cycle_schedules
+  has_many :order_cycle_shipping_methods
+  has_many :shipping_methods, class_name: "Spree::ShippingMethod",
+                              through: :order_cycle_shipping_methods
   has_paper_trail meta: { custom_data: proc { |order_cycle| order_cycle.schedule_ids.to_s } }
 
   attr_accessor :incoming_exchanges, :outgoing_exchanges
 
+  attribute :validate_shipping_methods, :boolean, default: true
```

Comment:
This increases code complexity a lot. Would it be possible to call the depending validations manually when needed?